File: src/main/java/net/tarks/craftingmod/nccauth/Comment.java
package net.tarks.craftingmod.nccauth;

import java.time.Instant;

public class Comment {
    public String userID;
    public String nickname;
    public String content;
    public Instant timestamp;
    public Comment(String uID,String nick,String con,Instant ts){
        userID = uID;
        nickname = nick;
        content = con;
        timestamp = ts;
    }
    public Comment(){
        userID = null;
        nickname = null;
        content = null;
        timestamp = Instant.ofEpochMilli(0);
    }
    public long getTimeDelta(){
        return Instant.now().getEpochSecond() - timestamp.getEpochSecond();
    }
}

File: node/src/structure/comment.ts
export default class Comment {
    public content:string;
    public userid:string;
    public nickname:string;
    public timestamp:number;

    public constructor(con:any) {
        this.content = con.content;
        this.userid = con.userid;
        this.nickname = con.nickname;
        this.timestamp = (con.timestamp == null) ? Date.now() : con.timestamp;
    }
    public getTimeDiffer() {
        return new Date().getTime() - this.timestamp;
    }
}
